horace_core/sqw/@sqw/private/do_sqw_eval_filebacked_.m

```diff
@@ -22,57 +22,63 @@
 % whilst avoiding an extra loop over pixels.
 %
 %==============================================================================
-%
-%Re #928 this function should be fixed. It does not work any more (and did not work before the changes anyway)
 
-pg_size = wout.pix.base_page_size;
-pix = wout.pix;
+pg_size = get(hor_config, 'mem_chunk_size');
+write_as_sqw = ~isempty(outfile);
+
 npix = wout.data.npix;
 
-% divide npix into chunks of the base page size
+% divide npix into chunks of the page size
 [npix_chunks, idxs] = split_vector_fixed_sum(npix(:), pg_size);
 
 % write dnd data to output file as per header description
 % this will be overwritten later with the recalculated image
-ldr = write_sqw_no_pix_or_footers_(wout, outfile);
-ldr_clob = onCleanup(@() ldr.delete());
+if write_as_sqw
+    ldr = write_sqw_no_pix_or_footers_(wout, outfile);
+    ldr_clob = onCleanup(@() ldr.delete());
+else
+    wout.pix = wout.pix.get_new_handle();
+end
 
 img_signal = zeros(1, numel(npix));
-for chunk_num = 1:numel(npix_chunks)
+
+
+s_ind = wout.pix.check_pixel_fields('signal');
+v_ind = wout.pix.check_pixel_fields('variance');
+
+for i = 1:wout.pix.num_pages
+    [wout.pix, data] = wout.pix.load_page(i);
+    npix_chunk = npix_chunks{i};
+    idx_chunk = idxs(:, i);
+
     qw = calculate_qw_pixels(wout);
     sig_chunk = sqwfunc(qw{:}, pars{:});
 
-    ldr = write_pix_chunk_with_signal_(ldr, pix, sig_chunk);
-
-    img_signal = increment_signal_sums_( ...
-        img_signal, sig_chunk, npix_chunks{chunk_num}, idxs(:, chunk_num) ...
-    );
+    data(s_ind, :) = sig_chunk;
```

Comment:
If written like this, averages are not recalculated

Reply:
I'm not sure what you mean by this. What averages? How should they be calculated?